packages/oauth/oauth-provider/src/token/token-manager.ts

```diff
@@ -477,41 +430,65 @@ export class TokenManager {
     }
   }
 
+  public isTokenExpired(client: Client, { data }: TokenInfo): boolean {
+    // @TODO This value should be computable even if we don't have the "client"
+    // (because fetching client info could be flaky). Instead, all the info
+    // needed should be stored in the token info.
+    const lifetime =
+      data.clientAuth.method === 'none' && !client.info.isFirstParty
+        ? UNAUTHENTICATED_REFRESH_LIFETIME
+        : AUTHENTICATED_REFRESH_LIFETIME
+    return data.createdAt.getTime() + lifetime < Date.now()
+  }
+
+  public isTokenInactive(client: Client, { data }: TokenInfo): boolean {
+    // @TODO This value should be computable even if we don't have the "client"
+    // (because fetching client info could be flaky). Instead, all the info
+    // needed should be stored in the token info.
+    const inactivityTimeout =
+      data.clientAuth.method === 'none' && !client.info.isFirstParty
+        ? UNAUTHENTICATED_REFRESH_INACTIVITY_TIMEOUT
+        : AUTHENTICATED_REFRESH_INACTIVITY_TIMEOUT
+    return data.updatedAt.getTime() + inactivityTimeout < Date.now()
+  }
+
   /**
    * @see {@link https://datatracker.ietf.org/doc/html/rfc7009#section-2.2 | RFC7009 Section 2.2}
    */
-  async revoke(token: string): Promise<void> {
+  async revoke(token: string): Promise<undefined | TokenInfo> {
     switch (true) {
       case isTokenId(token): {
```

Comment:
Not directly related to this work, but the `switch (true)` is a little bit of an odd construction.  No love for the good ole `if`s 😆 ?

Reply:
Haha I love this construct. But I can change to ifs.